testsuite/smoke-test/src/operational_tooling.rs

```diff
@@ -291,71 +245,121 @@ fn test_network_key_rotation_recovery() {
-    client_proxy_0.set_accounts(client_proxy_1.copy_all_accounts());
-    client_proxy_1
-        .mint_coins(&["mintb", "0", "10", "Coin1"], true)
+fn test_operator_key_rotation() {
+    let mut swarm = SmokeTestEnvironment::new(5);
+    swarm.validator_swarm.launch();
+
+    // Load a node config
+    let node_config =
+        NodeConfig::load(swarm.validator_swarm.config.config_files.first().unwrap()).unwrap();
+
+    // Connect the operator tool to the first node's JSON RPC API
+    let op_tool = OperationalTool::new(
+        format!("http://127.0.0.1:{}", node_config.rpc.address.port()),
+        ChainId::test(),
+    );
+
+    // Load validator's on disk storage
+    let backend = load_backend_storage(&node_config);
+
+    let (txn_ctx, _) = op_tool.rotate_operator_key(&backend).unwrap();
+    let mut client = swarm.get_validator_client(0, None);
+    client
+        .wait_for_transaction(txn_ctx.address, txn_ctx.sequence_number + 1)
         .unwrap();
-    client_proxy_1
-        .wait_for_transaction(treasury_compliance_account_address(), 1)
+
+    // Verify that the transaction was executed correctly
+    let result = op_tool
+        .validate_transaction(txn_ctx.address, txn_ctx.sequence_number)
         .unwrap();
-    assert!(compare_balances(
-        vec![(10.0, "Coin1".to_string())],
-        client_proxy_1.get_balances(&["b", "0"]).unwrap(),
-    ));
-    // wait for the mint txn in node 0
-    client_proxy_0
-        .wait_for_transaction(testnet_dd_account_address(), 1)
+    let vm_status = result.unwrap();
+    assert_eq!(VMStatusView::Executed, vm_status);
+
+    // Rotate the consensus key to verify the operator key has been updated
+    let (txn_ctx, new_consensus_key) = op_tool.rotate_consensus_key(&backend).unwrap();
+    let mut client = swarm.get_validator_client(0, None);
+    client
+        .wait_for_transaction(txn_ctx.address, txn_ctx.sequence_number + 1)
         .unwrap();
-    assert!(compare_balances(
-        vec![(10.0, "Coin1".to_string())],
-        client_proxy_0.get_balances(&["b", "0"]).unwrap(),
-    ));
-    let peer_id = env.get_validator(0).unwrap().validator_peer_id().unwrap();
-    let op_tool = env.get_op_tool(1);
-    let libra_root = load_libra_root_storage(node_configs.first().unwrap());
-    let context = op_tool.remove_validator(peer_id, &libra_root).unwrap();
-    client_proxy_1
-        .wait_for_transaction(context.address, context.sequence_number + 1)
+
+    // Verify that the config has been updated correctly with the new consensus key
+    let validator_account = node_config.validator_network.as_ref().unwrap().peer_id();
+    let config_consensus_key = op_tool
+        .validator_config(validator_account, &backend)
+        .unwrap()
+        .consensus_public_key;
+    assert_eq!(new_consensus_key, config_consensus_key);
+}
+
+#[test]
+fn test_operator_key_rotation_recovery() {
+    let mut swarm = SmokeTestEnvironment::new(5);
```

Comment:
why do we need 5?

Reply:
This is legacy, but a good point: I'd like to go through all of our smoke tests and check for this because it doesn't seem like things are consistent across all of our code. Will hold off on a refactor in this PR and punt to an upcoming one!